tests/imputers/test_BaseImputer.py

```diff
@@ -2,202 +2,268 @@
-
-        ta.classes.test_object_attributes(
-            obj=transformer,
-            expected_attributes={"impute_values_": impute_values},
-            msg="impute_values_ changed in transform",
-        )
-
-    def expected_df_1():
-        """Expected output of test_expected_output_1."""
-        df = pd.DataFrame(
-            {
-                "a": [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0],
-                "b": ["a", "b", "c", "d", "e", "f", np.nan],
-                "c": ["a", "b", "c", "d", "e", "f", np.nan],
-            },
-        )
-
-        df["c"] = df["c"].astype("category")
-
-        return df
-
-    def expected_df_2():
-        """Expected output of test_expected_output_2."""
-        df2 = pd.DataFrame(
-            {
-                "a": [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, np.nan],
-                "b": ["a", "b", "c", "d", "e", "f", "g"],
-                "c": ["a", "b", "c", "d", "e", "f", np.nan],
-            },
-        )
+        transformer.transform(create_df_1)
 
-        df2["c"] = df2["c"].astype("category")
-
-        return df2
-
-    def expected_df_3():
-        """Expected output of test_expected_output_3."""
-        df3 = pd.DataFrame(
-            {
-                "a": [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, np.nan],
-                "b": ["a", "b", "c", "d", "e", "f", "g"],
-                "c": ["a", "b", "c", "d", "e", "f", "f"],
-            },
-        )
-
-        df3["c"] = df3["c"].astype("category")
-
-        return df3
+        assert (
+            transformer.impute_values_ == impute_values
+        ), "impute_values_ changed in transform"
 
     @pytest.mark.parametrize(
-        ("df", "expected"),
-        ta.pandas.adjusted_dataframe_params(d.create_df_2(), expected_df_1()),
+        ("library", "expected_df_1"),
+        [("pandas", "pandas"), ("polars", "polars")],
+        indirect=["expected_df_1"],
     )
-    def test_expected_output_1(self, df, expected, initialized_transformers):
+    def test_expected_output_1(self, library, expected_df_1, initialized_transformers):
         """Test that transform is giving the expected output when applied to float column."""
-        x1 = initialized_transformers[self.transformer_name]
-        x1.impute_values_ = {"a": 7}
-        x1.columns = ["a"]
+        # Create the DataFrame using the library parameter
+        create_df_2 = d.create_df_2(library=library)
 
-        df_transformed = x1.transform(df)
-
-        ta.equality.assert_equal_dispatch(
-            expected=expected,
-            actual=df_transformed,
-            msg=f"Error from {self.transformer_name} transform col a",
-        )
+        # Initialize the transformer
+        transformer = initialized_transformers[self.transformer_name]
+        transformer.impute_values_ = {"a": 7}
+        transformer.columns = ["a"]
+
+        # Transform the DataFrame
+        df_transformed = transformer.transform(create_df_2)
+
+        # Check outcomes for single rows
```

Comment:
S: on tests like this where you look at single rows, could you also do and assert_frame_equal on the fulls dfs at the end? just to cover everything